metricbeat/mb/testing/data_test.go

```diff
@@ -0,0 +1,181 @@
+	"github.com/elastic/beats/metricbeat/mb"
+
+	// TODO: generate include file for these tests automatically moving forward
+	_ "github.com/elastic/beats/metricbeat/module/kibana/status"
+	_ "github.com/elastic/beats/metricbeat/module/rabbitmq/connection"
+	_ "github.com/elastic/beats/metricbeat/module/traefik/health"
+)
+
+var (
+	// Use `go test -generate` to update files.
+	generateFlag = flag.Bool("generate", false, "Write golden files")
+)
+
+type Config struct {
+	Type string
+	URL  string
+}
+
+func TestAll(t *testing.T) {
+
+	configFiles, _ := filepath.Glob("../../module/*/*/_meta/testdata/config.yml")
+
+	for _, f := range configFiles {
+		// get module and metricset name from path
+		s := strings.Split(f, "/")
+		moduleName := s[3]
+		metricSetName := s[4]
+
+		configFile, err := ioutil.ReadFile(f)
+		if err != nil {
+			log.Printf("yamlFile.Get err   #%v ", err)
+		}
+		var config Config
+		err = yaml.Unmarshal(configFile, &config)
+		if err != nil {
+			log.Fatalf("Unmarshal: %v", err)
+		}
+
+		getTestdataFiles(t, config.URL, moduleName, metricSetName)
+	}
+}
+
+func getTestdataFiles(t *testing.T, url, module, metricSet string) {
+
+	ff, _ := filepath.Glob("../../module/" + module + "/" + metricSet + "/_meta/testdata/*.json")
+	var files []string
+	for _, f := range ff {
+		// Exclude all the expected files
+		if strings.HasSuffix(f, "-expected.json") {
+			continue
+		}
+		files = append(files, f)
+	}
+
+	for _, f := range files {
+		t.Run(f, func(t *testing.T) {
+			runTest(t, f, module, metricSet, url)
+		})
+	}
+}
+
+func runTest(t *testing.T, file string, module, metricSetName, url string) {
+
+	// starts a server serving the given file under the given url
+	s := server(t, file, url)
+	defer s.Close()
+
+	metricSet := NewReportingMetricSetV2(t, getConfig(module, metricSetName, s.URL))
+	events, errs := ReportingFetchV2(metricSet)
+
+	// Gather errors to build also error events
+	for _, e := range errs {
+		// TODO: for errors strip out and standardise the URL error as it would create a different diff every time
+		events = append(events, mb.Event{Error: e})
+	}
+
+	var data []common.MapStr
+
+	for _, e := range events {
+		beatEvent := StandardizeEvent(metricSet, e, mb.AddMetricSetInfo)
+		// Overwrite service.address as the port changes every time
+		beatEvent.Fields.Put("service.address", "127.0.0.1:55555")
+		data = append(data, beatEvent.Fields)
+	}
+
+	output, err := json.MarshalIndent(&data, "", "    ")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Overwrites the golden files if run with -generate
+	if *generateFlag {
+		if err = ioutil.WriteFile(file+"-expected.json", output, 0644); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Read expected file
+	expected, err := ioutil.ReadFile(file + "-expected.json")
+	if err != nil {
+		t.Fatalf("could not read file: %s", err)
+	}
+
+	assert.Equal(t, string(expected), string(output))
+
+	if strings.HasSuffix(file, "docs.json") {
+		writeDataJSON(t, data[0], module, metricSetName)
```

Comment:
It'd be nice to support here something to select events too, like #10367.

Reply:
I think we don't need this anymore because we have the docs.json convention which generates the data.json. So if we have a specific event that looks interesting, we should put it into the docs.json without any other events.